src/types.ts

```diff
@@ -16,99 +19,198 @@ export interface ConfigInterface<
   revalidateOnMount?: boolean
   revalidateOnReconnect: boolean
   shouldRetryOnError: boolean
-  fetcher: Fn
   suspense: boolean
+  fetcher: Fn
   initialData?: Data
 
   isOnline: () => boolean
   isDocumentVisible: () => boolean
   isPaused: () => boolean
-  onLoadingSlow: (key: string, config: ConfigInterface<Data, Error>) => void
+  onLoadingSlow: (key: string, config: Configuration<Data, Error>) => void
```

Comment:
would you also want to add the `?` back to mark those properties as optional again in this PR?

Reply:
This is the internal type which is more strict. The exported one has `Partial` so all properties are optional.